XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/TeleportSystem/FadingTeleportHandler.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.Rendering;
+using XRTK.EventDatum.Teleport;
+using XRTK.Extensions;
+using XRTK.Interfaces.TeleportSystem;
+
+namespace XRTK.SDK.TeleportSystem
+{
+    /// <summary>
+    /// This <see cref="IMixedRealityTeleportSystem"/> handler implementation will
+    /// fade out the camera when teleporting and fade it back in when done.
+    /// </summary>
+    [System.Runtime.InteropServices.Guid("0db5b0fd-9ac3-487a-abfd-754963f4e2a3")]
+    public class FadingTeleportHandler : BaseTeleportHandler
+    {
+        [SerializeField]
+        [Tooltip("Assign the transform with the camera component attached. If not set, the component uses its own transform.")]
+        private Transform cameraTransform = null;
+
+        [SerializeField]
+        [Tooltip("Assign the transform being teleported to the target location. If not set, the component game object's parent transform is used.")]
+        private Transform teleportTransform = null;
```

Comment:
I feel like the camera transform and telelport transform could be defined in the profile.

Reply:
Not sure how. Those fields are part of the handler implementation. So the profile inspector would have to "adjust" depending on the selected handler impelmenation.

Reply:
I don't think we need to do that. I setup the handlers in a way that they can auto-configure reasonably. And since they get attached to the camera GameObject people can adjust there, if needed. The teleport system will not override any existing configuration.

Reply:
We do something similar for each of the platform specific profiles. If we need to change or adjust the inspector to show additional fields we can.

Also I was wondering if these fields would better belong in the base teleport handler since each implementation seems to need to know the teleport and camera transforms.